### PR TITLE
go.mod: remove the package dependency of github.com/pingcap/tiup

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Make TiDB 4.0 use the PD 4.0 branch.

**Our current dependency management is totally a mess.**

`tidb` depends on `pd/client`
`br` also depends on `tidb` and `pd`
But they depend on a different versions of `pd`... after I manually update the `br` in tidb go.mod, it seems to work.

The other problem is `tiup`, I'm not sure why `pd` depend on `tiup`
`tiup` depends on `github.com/tikv/pd v1.1.0-beta.0.20200824114021-f8c45ae287fd`
And if tidb imports `pd`, it has to use `f8c45ae287fd`

Here is the error message:

```
genius@genius-xps:/media/genius/OS/project/src/github.com/pingcap/tidb$ go get github.com/tikv/pd@9e238907894ffd3600ceed70e2749df1f64ecdbe
go: github.com/tikv/pd 9e238907894ffd3600ceed70e2749df1f64ecdbe => v0.0.0-20201022093918-9e238907894f
go get: inconsistent versions:
        github.com/tikv/pd@v0.0.0-20201022093918-9e238907894f from github.com/tikv/pd@9e238907894ffd3600ceed70e2749df1f64ecdbe requires github.com/tikv/pd@v1.1.0-beta.0.20200824114021-f8c45ae287fd (not github.com/tikv/pd@v0.0.0-20201022093918-9e238907894f from github.com/tikv/pd@9e238907894ffd3600ceed70e2749df1f64ecdbe)
```

### What is changed and how it works?

PD should not import `github.com/pingcap/tiup`, it's not reasonable.
TiDB must use the latest pd cliet to fix a serious bug (tsloop hang).

### Check List

Tests

- No code

Code changes

- Has configuration change
- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tidb-ansible`](https://github.com/pingcap/tidb-ansible):
- Need to cherry-pick to the release branch

### Release note

No release note

### CODE REVIEW

**Review by rleungx — APPROVED** [review]

**Comment by HunDunDM** [comment]
The PD of the current `master` branch does not depend on `tiup`. Does the `release-4.0` branch lack some related cherry-pick? In addition, this PR introduces too many indirect dependencies.

**Review by HunDunDM — COMMENTED** [review]
The PD of the current master branch does not depend on tiup. Does the release-4.0 branch lack some related cherry-pick? In addition, this PR introduces too many indirect dependencies.

**Comment by tiancaiamao** [comment]
> The PD of the current `master` branch does not depend on `tiup`. Does the `release-4.0` branch lack some related cherry-pick? 

Maybe the author does not aware of that, maybe it's not a bug fix PR or something that seems important.
But it takes me a lot of time to figure out why tidb 4.0 can not use the pd 4.0, the package dependency is really a mess, you know.

> In addition, this PR introduces too many indirect dependencies.

I've removed the `indirect` dependency manually and run `go mod tidy`. @HunDunDM

File: go.mod [hunk 1]
```diff
@@ -40,6 +40,7 @@ require (
 	github.com/syndtr/goleveldb v1.0.1-0.20190318030020-c3a204f8e965
 	github.com/unrolled/render v1.0.1
 	github.com/urfave/negroni v0.3.0
+	go.etcd.io/bbolt v1.3.4 // indirect
```
Comment (tiancaiamao) [inline go.mod:43]: If we let `go`  choose the module, it use v1.3.3 and CI fail with this error:

```
fatal error: checkptr: unsafe pointer conversion

goroutine 182 [running]:
runtime.throw(0x2d570fe, 0x23)
        /opt/hostedtoolcache/go/1.14.13/x64/src/runtime/panic.go:1116 +0x72 fp=0xc000ba16a8 sp=0xc000ba1678 pc=0xc12a12
runtime.checkptrAlignment(0xc000bac9d0, 0x2c15ae0, 0x1)
        /opt/hostedtoolcache/go/1.14.13/x64/src/runtime/checkptr.go:20 +0xc9 fp=0xc000ba16d8 sp=0xc000ba16a8 pc=0xbe37f9
go.etcd.io/bbolt.(*Bucket).write(0xc000ba1850, 0x0, 0x0, 0x0)
        /home/runner/go/pkg/mod/go.etcd.io/bbolt@v1.3.3/bucket.go:624 +0x15c fp=0xc000ba1740 sp=0xc000ba16d8 pc=0x17d2d2c
go.etcd.io/bbolt.(*Bucket).CreateBucket(0xc000bb4018, 0x4801374, 0x7, 0x7, 0x5555555555555555, 0x38, 0x8)
        /home/runner/go/pkg/mod/go.etcd.io/bbolt@v1.3.3/bucket.go:181 +0x465 fp=0xc000ba1900 sp=0xc000ba1740 pc=0x17cf175
go.etcd.io/bbolt.(*Tx).CreateBucket(...)
        /home/runner/go/pkg/mod/go.etcd.io/bbolt@v1.3.3/tx.go:108
go.etcd.io/etcd/mvcc/backend.(*batchTx).UnsafeCreateBucket(0xc000b84690, 0x4801374, 0x7, 0x7)
        /home/runner/go/pkg/mod/go.etcd.io/etcd@v0.5.0-alpha.5.0.20191023171146-3cf2f69b5738/mvcc/backend/batch_tx.go:72 +0x9c fp=0xc000ba1b30 sp=0xc000ba1900 pc=0x17ffffc
go.etcd.io/etcd/mvcc/backend.(*batchTxBuffered).UnsafeCreateBucket(0xc000b84690, 0x4801374, 0x7, 0x7)
        <autogenerated>:1 +0x62 fp=0xc000ba1b60 sp=0xc000ba1b30 pc=0x180bff2
go.etcd.io/etcd/etcdserver/api/membership.mustCreateBackendBuckets(0x37907c0, 0xc000b96600)
```

**Pull request closed without merging**